src/components/MultiSelect/MultiSelectDialog.tsx

```diff
@@ -113,121 +156,299 @@ export const MultiSelectDialog = chakra(
-            <MultiSelectMenuButton
-              id={`ms-${id}-menu-button`}
-              multiSelectId={id}
-              multiSelectLabelText={labelText}
-              isOpen={isOpen}
-              selectedItems={selectedItems}
-              onMenuToggle={() => {
-                setIsOpen(!isOpen);
-              }}
-              onClear={onClear}
-            />
-            <Box
-              role="dialog"
-              __css={styles.menuContainer}
-              {...(isOpen && { "aria-modal": true })}
-              aria-labelledby={`ms-${id}-menu-button`}
-            >
-              <UnorderedList
-                styleType="none"
-                marginInlineStart="0"
-                __css={styles.menu}
+      const handleSearchKeyword = (event) => {
+        const value = event.target.value.trim().toLowerCase();
+
+        if (value === "") {
+          return listItemsCount === defaultItemsVisible
+            ? displayDefaultItems()
+            : setItemsList(items);
+        }
+
+        const filteredItems = items.filter((item) => {
+          if (item.children) {
+            if (item.name.toLowerCase().includes(value)) {
+              return true;
+            }
+
+            return item.children.some((childItem) =>
+              childItem.name.toLowerCase().includes(value)
+            );
+          }
+
+          return item.name.toLowerCase().includes(value);
+        });
+
+        setItemsList(filteredItems);
+      };
+
+      const clearSearchKeyword = () => {
+        listItemsCount === defaultItemsVisible
+          ? displayDefaultItems()
+          : setItemsList(items);
+      };
+
+      const displayDefaultItems = () => {
+        const list = [];
+        let count = 0;
+
+        for (let i = 0; i < items.length && count < defaultItemsVisible; i++) {
+          const currentItem = { ...items[i] };
+
+          count++;
+
+          if (currentItem.children) {
+            const showChilds = isAllChecked(id, currentItem);
+            const showIndeterminateChilds = isIndeterminate(id, currentItem);
+            setDefaultCheckboxes("block");
+
+            if (!showChilds && !showIndeterminateChilds) {
+              currentItem.children.forEach((childItem) => {
+                if (count < defaultItemsVisible) {
+                  count++;
+                  // Display the first defaultItemsVisible children
+                  setDefaultChildCheckboxes((prev) => {
+                    // Assuming 'prev' is an array, and we want to update it by adding 'childItem'
+                    return [...prev, { ...childItem, display: "block" }];
+                  });
+                } else {
+                  // Hide the remaining children
+                  setDefaultChildCheckboxes((prev) => {
+                    // Assuming 'prev' is an array, and we want to update it by adding 'childItem'
+                    return [...prev, { ...childItem, display: "none" }];
```

Comment:
I don't understand what this is trying to do. When I log this, I get this which seems wrong:
![Screen Shot 2023-12-12 at 12 26 32 PM](https://github.com/NYPL/nypl-design-system/assets/1280564/e0760072-4f30-467e-9f0a-f360708856f9)

Notice that "none" is being add individually as letters in the array.

Reply:
There's a bigger issue. `defaultChildCheckboxes` is currently defined as a string. You have it as:
```
const [defaultChildCheckboxes, setDefaultChildCheckboxes] = useState("none");
```

When you pass `defaultChildCheckboxes` to the `Checkbox` component, you're passing it as `display={defaultChildCheckboxes}`. This is fine.

The problem is that here you are updating `defaultChildCheckboxes` to an array instead of a string. So this needs to be reworked.